Analysis/Core/include/Analysis/TrackSelection.h

```diff
@@ -17,33 +17,39 @@
 
 #include "Framework/Logger.h"
 #include "Framework/DataTypes.h"
-#include "TObject.h"
 #include <set>
+#include "TObject.h"
+#include "TFormula.h"
 
 class TrackSelection : public TObject
 {
-
  public:
-  TrackSelection();
+  TrackSelection() = default;
+  // explicitly disallow copying this object so pointer members are always handled porperly
+  TrackSelection(TrackSelection&&) = default;
+  TrackSelection(const TrackSelection&) = delete;
+  TrackSelection& operator=(TrackSelection&&) = default;
+  TrackSelection& operator=(const TrackSelection&) = delete;
 
   // Temporary function to check if track passes selection criteria. To be
   // replaced by framework filters
   template <typename T>
   bool IsSelected(T const& track)
   {
     if (track.trackType() == mTrackType &&
-        track.pt() >= mMinPt && track.pt() < mMaxPt && track.eta() >= mMinEta &&
-        track.eta() < mMaxEta && track.tpcNClsFound() >= mMinNClustersTPC &&
+        track.pt() >= mMinPt && track.pt() <= mMaxPt && track.eta() >= mMinEta &&
+        track.eta() <= mMaxEta && track.tpcNClsFound() >= mMinNClustersTPC &&
         track.tpcNClsCrossedRows() >= mMinNCrossedRowsTPC &&
         track.tpcCrossedRowsOverFindableCls() >=
           mMinNCrossedRowsOverFindableClustersTPC &&
         (track.itsNCls() >= mMinNClustersITS) &&
-        (track.itsChi2NCl() < mMaxChi2PerClusterITS) &&
-        (track.tpcChi2NCl() < mMaxChi2PerClusterTPC) &&
+        (track.itsChi2NCl() <= mMaxChi2PerClusterITS) &&
+        (track.tpcChi2NCl() <= mMaxChi2PerClusterTPC) &&
         ((mRequireITSRefit) ? (track.flags() & 0x4) : true) &&
         ((mRequireTPCRefit) ? (track.flags() & 0x40) : true) &&
         FulfillsITSHitRequirements(track.itsClusterMap()) &&
-        abs(track.dcaXY()) < mMaxDcaXY && abs(track.dcaZ()) < mMaxDcaZ) {
+        abs(track.dcaXY()) <= ((mMaxDcaXYPtDep) ? mMaxDcaXYPtDep->Eval(track.pt()) : mMaxDcaXY) &&
```

Comment:
I am worried that the Eval will consume quite some computing resources, and to my knowledge this pT dependent cut has hardly changed throughout Run 2. What do you think to instead hard code the formula, but allow the possibility to put other formulas in the future.
E.g. by having an enum for mMaxDcaXYPtDepFormula which then can map to several functions.

Reply:
@aalkin @ktf What do you think?

Reply:
How would you hard-code a formula? A map with different values for different pt-values/intervals? Not sure if this would be more efficient.

Reply:
I assume `Eval` is from ROOT's TF1? Having a lambda instead of TF1 may help.

Reply:
its from TFormula

Reply:
I will give the lambda a try. Thanks for the hint!